### DISCUSSION TITLE
Not making any trades

### DISCUSSION THREAD

**Discussion by alizain5693** [kernc/backtesting.py#411] [Q&A] [answered]
I'm probably missing something obvious but this strategy isn't making any trades. I'm getting buy and sell signals but it doesn't actually do a trade. please help

### Expected Behavior

Indicators were precalculated and added to the data frame. I expected trades to be made based on those indicators

### Actual Behavior

It didn't make any trades. I also didn't get any errors.

### Steps to Reproduce

Use the code below and the data frame from the linked csv file.

Link: https://github.com/alizain5693/data/blob/main/ethereumData.csv

### Additional info
`
# strategy

    
    class EMAcross(Strategy):
      profit = 2
      loss = 1
  
      def init(self):
  
      
      def next(self):
          # If sma1 crosses above sma2, close any existing
          # short trades, and buy the asset
          if crossover(self.data.Close, self.data.EMA):
              if((self.data.Hammer>0) or (self.data.InvertedHammer>0) or (self.data.Engulfing>0) or (self.data.Piercing>0) or(self.data.MorningStar>0) or (self.data.WhiteSoldiers>0) ):
              # if(self.data.Hammer>0):
                  # success = self.data.Close * (1+(self.profit)/100)
                  # fail = self.data.Close * (1-(self.loss)/100)
                  self.position.close()
                  self.buy()
                  # self.buy(sl = fail, tp = success)   
                  print("BUY")
              else:
                  pass
  
          # Else, if sma1 crosses below sma2, close any existing
          # long trades, and sell the asset
          elif crossover(self.data.EMA, self.data.Close):
              if((self.data.HangingMan<0) or (self.data.ShootingStar<0) or (self.data.Engulfing<0) or (self.data.EveningStar<0) or (self.data.BlackCrows<0) or (self.data.DarkCloudCover<0)):   
              # if(self.data.HangingMan<0):
                  # success = self.data.Close * (1-(self.profit)/100)
                  # fail = self.data.Close * (1+(self.loss)/100)
                  self.position.close()
                  self.sell()
                  # self.sell(sl = fail, tp = success)
                  print("SELL")
              else:
                  pass

  bt = Backtest(data,EMAcross,cash = 1000, commission = 0.00036)
  stats = bt.run()
  print("INITIAL STATS")
  print(stats)
  print("INITIAL PLOT")
  bt.plot()

`

- Latest version

**Comment by sersalek** [comment, marked as answer]
Probably solved by now, but I have to let you know that if the initial investment is less than the prices, no trade will happen.

**Reply by MrDenfish** [reply]
Thanks for stating the obvious, it helped me with the same problem.

“There is nothing more deceptive than an obvious fact.” 
S.Holmes
